src/CMakeLists.txt

```diff
@@ -247,6 +247,9 @@ set_target_properties(parthenon PROPERTIES SOVERSION ${parthenon_VERSION})
 
 target_compile_features(parthenon PUBLIC cxx_std_17)
 
+target_link_options(parthenon PUBLIC "$<$<AND:$<CXX_COMPILER_ID:GNU>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.1>>:-lstdc++fs>" )
+target_link_options(parthenon PUBLIC "$<$<AND:$<CXX_COMPILER_ID:Clang>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.0>>:-lc++fs>" )
```

Comment:
This change was introduced to support backwards compatibility for `std::format`, which was introduced in #848 . Neither @pgrete nor I noticed that 

> Using this library may require additional compiler/linker options. GNU implementation prior to 9.1 requires linking with -lstdc++fs and LLVM implementation prior to LLVM 9.0 requires linking with -lc++fs. 

I take responsibility. Wish I'd noticed the fine print.


This set of cmake generator expressions should enable older versions of gcc/clang to link appropriately. Probably should be tested more carefully but it seems to work for us locally.

Reply:
This doesn't compile under NVHPC, and probably won't under many compiler IDs we otherwise support (IntelLLVM != Clang, AOCC != Clang, IBM, etc, see [list](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html)).

Would it be possible to just try to link both libraries, and ignore failures?

Reply:
After messing with `find_library` a while, I've decided that libraries are hard even if you don't special-case compilers, and it would be much easier to use something like `find_package(Filesystem)`, i.e., which someone else has already written.

Luckily these exist, I've had good luck with [this one](https://github.com/lethal-guitar/RigelEngine/blob/master/cmake/Modules/FindFilesystem.cmake), released under BSD license.  (There is a rather hiliarously old CMake [bug](https://gitlab.kitware.com/cmake/cmake/-/issues/17834) to make something like this standard). Happy to add the file and the couple of lines required to switch to this, if I get the okay.

Reply:
:+1: I'm in favor of this, and thanks for finding it! Please feel free to push directly to this branch.

Reply:
> Is testing older GCC's in the CI worth it to catch these sorts of issues? Probably not?

It might be worth it... but the problem will just go away with time.